core/templates/pages/exploration-editor-page/settings-tab/settings-tab.component.spec.ts

```diff
@@ -255,33 +255,42 @@ describe('Settings Tab Component', () => {
     });
 
     it('should initialize controller properties after its initialization',
-      () => {
+      (done) => {
         expect(ctrl.isRolesFormOpen).toBe(false);
         expect(ctrl.canDelete).toBe(true);
         expect(ctrl.canModifyRoles).toBe(true);
         expect(ctrl.canReleaseOwnership).toBe(true);
         expect(ctrl.canUnpublish).toBe(true);
         expect(ctrl.explorationId).toBe(explorationId);
         expect(ctrl.canManageVoiceArtist).toBe(true);
+        expect(ctrl.loggedInUser).toBe('username1');
 
-        expect(ctrl.CATEGORY_LIST_FOR_SELECT2[0]).toEqual({
-          id: 'Astrology',
-          text: 'Astrology'
-        });
+        setTimeout(()=>{
```

Comment:
Is it possible to use `tick` instead of `setTimeout`?

Reply:
I have tried to use `tick` but it doesn't seem to work for this particular test.

Reply:
Have you tried writing the test similar to [this file](https://github.com/oppia/oppia/blob/f74b606e8511cd4296b3c99aad37e53b66cca196/core/templates/pages/exploration-editor-page/modal-templates/editor-reloading-modal.component.spec.ts#L62-L72
)?

Reply:
Yup, it still doesn't work.
![image](https://user-images.githubusercontent.com/73544247/169527206-24062153-58dc-433e-81c4-c124da53e8ea.png)

Reply:
Have you tried placing the `tick` after calling `$scope.$apply();`?

Reply:
@gp201 I tried calling $scope.$apply() before tick, it doesn't seem to work as well.